tests/integration_tests/functional/test_snapshot_basic.py

```diff
@@ -0,0 +1,182 @@
+                                                     tapname="tap0")
+    logger.debug("Host IP: {}, Guest IP: {}".format(host_ip, guest_ip))
+
+    # We will need netmask_len in build_from_snapshot() call later.
+    netmask_len = network_config.get_netmask_len()
+    basevm.start()
+    ssh_connection = net_tools.SSHConnection(basevm.ssh_config)
+
+    # Verify if guest can run commands.
+    exit_code, _, _ = ssh_connection.execute_command("sync")
+    assert exit_code == 0
+
+    logger.info("Create {} #0.".format(snapshot_type))
+    # Create a snapshot builder from a microvm.
+    snapshot_builder = SnapshotBuilder(basevm)
+
+    # Create base snapshot.
+    snapshot = snapshot_builder.create([rw_disk],
+                                       ssh_key,
+                                       snapshot_type)
+
+    base_snapshot = snapshot
+    basevm.kill()
+
+    for i in range(5):
+        logger.info("Load snapshot #{}, mem {}".format(i, snapshot.mem))
+        microvm, tap = vm_builder.build_from_snapshot(snapshot,
+                                                      host_ip,
+                                                      guest_ip,
+                                                      netmask_len,
+                                                      True,
+                                                      enable_diff_snapshots)
+
+        # Attemp to connect to resumed microvm.
+        ssh_connection = net_tools.SSHConnection(microvm.ssh_config)
+
+        # Start a new instance of fio on each iteration.
+        fio = """fio --filename=/dev/vda --direct=1 --rw=randread --bs=4k
+        --ioengine=libaio --iodepth=16 --runtime=10 --numjobs=4 --time_based
+        --group_reporting --name=iops-test-job --eta-newline=1 --readonly"""
+        ssh_cmd = "screen -L -Logfile /tmp/fio{} -dmS test{} {}"
+        ssh_cmd = ssh_cmd.format(i, i, fio)
+        exit_code, _, _ = ssh_connection.execute_command(ssh_cmd)
+
+        logger.info("Create snapshot #{}.".format(i + 1))
+
+        # Create a snapshot builder from the currently running microvm.
+        snapshot_builder = SnapshotBuilder(microvm)
+
+        snapshot = snapshot_builder.create([rw_disk],
+                                           ssh_key,
+                                           snapshot_type)
+
+        # If we are testing incremental snapshots we must merge the base with
+        # current layer.
+        if snapshot_type == SnapshotType.DIFF:
+            logger.info("Base: {}, Layer: {}".format(base_snapshot.mem,
+                                                     snapshot.mem))
+            snapshot.rebase_snapshot(base_snapshot)
+            # Update the base for next iteration.
+            base_snapshot = snapshot
+
+        microvm.kill()
+        del tap
+
+
+@pytest.mark.skipif(
+    platform.machine() != "x86_64",
+    reason="Not supported yet."
+)
+def test_5_full_snapshots(network_config,
```

Comment:
Test passes, but I see the following error printed a couple of times:
```
Exception ignored in: <bound method Tap.__del__ of <host_tools.network.Tap object at 0x7fadebf2ccc0>>
Traceback (most recent call last):
  File "/firecracker/tests/host_tools/network.py", line 329, in __del__
    self.name
  File "/firecracker/tests/framework/utils.py", line 192, in run_cmd
    no_shell=no_shell))
  File "/usr/lib/python3.6/asyncio/base_events.py", line 484, in run_until_complete
    return future.result()
  File "/firecracker/tests/framework/utils.py", line 132, in run_cmd_async
    raise ChildProcessError(output_message)
ChildProcessError:
[1755] Command:
ip netns exec ca39d4c6-e9f0-4f04-92b3-33a01c58424c ip link set tap0 down
[1755] stderr:
Cannot open network namespace "ca39d4c6-e9f0-4f04-92b3-33a01c58424c": No such file or directory
```

Reply:
This looks like a cleanup bug, does it happen if you run this in a clean container ?

Reply:
I've changed the order of the cleanup to `del tap` first and it now looks good on my side.

Reply:
I've removed the tap cleanup part that was triggering the issue since the tap si destroyed along with the namespace when the jailer cleanup is executed.